cloudsim-plus-examples/src/main/java/org/cloudsimplus/examples/power/PowerExample.java

```diff
@@ -298,7 +297,7 @@ private Host createPowerHost() {
             peList.add(new PeSimple(1000, new PeProvisionerSimple()));
         }
 
-        final PowerModel powerModel = new PowerModelLinear(MAX_POWER_WATTS_SEC, STATIC_POWER_PERCENT);
+        final PowerModelHost powerModel = new PowerModelHost(STATIC_POWER, MAX_POWER);
```

Comment:
I agree, we have to be careful here. My concern with this change was not the order of arguments (i.e. we can definitely keep max first, static second) but to change the semantics of the static argument from `STATIC_POWER_PERCENT` to an absolute value `STATIC_POWER`, mainly because it's a lot more intuitive.

There is another reason: When you model entities in a more abstract way (like I did in LEAF), you often have entities that do not really have a `MAX_POWER` but only an incremental power per X, e.g. W/MIPS for CPUs, W/(bit/s) for network, ... In these PowerModels it's simply not possible to define the static power as a fraction of the max power. So for consistency, I thought it may be cool if `STATIC_POWER` is an absolute value in this PowerModel too.

You are right, that the big problem here is that both arguments have the same data type. Still, I think it may be ok to go for `PowerModelHost(MAX_POWER, STATIC_POWER)` instead of `PowerModelHost(MAX_POWER, STATIC_POWER_PERCENT)` since it's an entirely new class anyway. `PowerModelLinear` is gone.

What do you think?

Reply:
> I agree, we have to be careful here. My concern with this change was not the order of arguments (i.e. we can definitely keep max first, static second) but to change the semantics of the static argument from `STATIC_POWER_PERCENT` to an absolute value `STATIC_POWER`, mainly because it's a lot more intuitive.

For sure, but the issue is because we are changing the meaning of the value for this parameter, which 
will cause big troubles for users anyway. 

> 
> There is another reason: When you model entities in a more abstract way (like I did in LEAF), you often have entities that do not really have a `MAX_POWER` but only an incremental power per X, e.g. W/MIPS for CPUs, W/(bit/s) for network, ... In these PowerModels it's simply not possible to define the static power as a fraction of the max power. So for consistency, I thought it may be cool if `STATIC_POWER` is an absolute value in this PowerModel too.

I see. So we need to find a way that at least, causes a compilation error when the user upgrades.
Maybe we could follow the approach in `UtilizationModelAbstract`, using a `Unit` attribute. That is an enum defining if the value is absolute or percentage. We could have some attribute like `Unit staticPowerUnit`.
Then, we could add a constructor that assumes the Unit as PERCENTAGE by default, and other one that takes a `Unit` parameter. When a percentage value is given for the `staticPower` parameter, we just compute and store the absolute value.

The enum should only be moved to its own file. 
Tell me if you see other alternative.

> 
> You are right, that the big problem here is that both arguments have the same data type. Still, I think it may be ok to go for `PowerModelHost(MAX_POWER, STATIC_POWER)` instead of `PowerModelHost(MAX_POWER, STATIC_POWER_PERCENT)` since it's an entirely new class anyway. `PowerModelLinear` is gone.
> 

This is another issue I've commented about the existing PowerModels. 
The new implementation considers that power consumption is always linear, which some research shows it's not.
I think we could make `PowerModelHost ` abstract and `PowerModelLinear` to extend it.
This way, there will be no impact on existing simulations of other researchers.

Reply:
So the idea behind the new `PowerModel` is that it does not only work on hosts - like the old ones - but can also describe the power usage of a `Datacenter`, `VM`, network equipment, ...

Because of that, implementations like `PowerModelLinear` don't make sense, because `PowerModel` implementations are always specific for the type of entity they describe. For this, I created `PowerModelHost`, which is a simple, linear implementation of such a `PowerModel` for a `Host` (maybe you are right and we make it abstract and rather call the implementation `PowerModelHostSimple`? The only thing is, that there is nothing to implement on this abstract class haha).

In any way, the class name change will throw a compiler error so people can have a look at the interface, no?